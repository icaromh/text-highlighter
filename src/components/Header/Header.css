.Header{
  background-color: #fff;
  position: relative;
}

.Header::after {
  content: '';
  bottom: 0;
  left: 0;
  width: 100%;
  position: absolute;
  display: block;
  box-shadow: 0 0 0 1px rgba(0,0,0,.1)
}

.Header__title {
  padding: 1rem;
  margin: 0;
  font-family: Verdana, Geneva, Tahoma, sans-serif;
  font-weight: 400;
}

