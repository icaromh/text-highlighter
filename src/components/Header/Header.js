import React from 'react'
import PropTypes from 'prop-types'

const Header = ({ title, children }) => (
  <div className="Header">
    <h1 className="Header__title">{title}</h1>
    {children}
  </div>
)

Header.propTypes = {
  title: PropTypes.string.isRequired,
  children: PropTypes.object.isRequired,
}

export default Header
