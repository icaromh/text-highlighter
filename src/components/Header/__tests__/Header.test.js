import React from 'react'
import { shallow } from 'enzyme'

import Header from '../Header'

describe('#Header', () => {
  test('expect to render', () => {
    const wrapper = shallow(
      <Header title="dummy">
        <p>some subtitle</p>
      </Header>,
    )
    expect(wrapper).toMatchSnapshot()
  })
})
