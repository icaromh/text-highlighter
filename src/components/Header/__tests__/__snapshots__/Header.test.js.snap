// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Header expect to render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    title="dummy"
  >
    <p>
      some subtitle
    </p>
  </Header>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="Header__title"
        >
          dummy
        </h1>,
        <p>
          some subtitle
        </p>,
      ],
      "className": "Header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "dummy",
          "className": "Header__title",
        },
        "ref": null,
        "rendered": "dummy",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "some subtitle",
        },
        "ref": null,
        "rendered": "some subtitle",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="Header__title"
          >
            dummy
          </h1>,
          <p>
            some subtitle
          </p>,
        ],
        "className": "Header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "dummy",
            "className": "Header__title",
          },
          "ref": null,
          "rendered": "dummy",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "some subtitle",
          },
          "ref": null,
          "rendered": "some subtitle",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
